### PR TITLE
Update `skaffold` to `v2.2.0`

### PR DESCRIPTION
**Pull request by oliver-goetz**
**How to categorize this PR?**

/area dev-productivity
/kind enhancement

**What this PR does / why we need it**:
This PR updates `skaffold` to its most recent version `v2.2.0`.
Skaffold `Config` has to be updated from `skaffold/v2beta29` to `skaffold/v4beta3` for this update.

This was done in a two step approach:
1. `skaffold fix ...`
2. Manual adjustments

On the positive side of these changes is that the `imageVectorOverwrite` directives could be simplified.

Alongside, it updates other tools:
```
- helm to v3.11.2
- kubectl to v1.24.11
- yq to v4.31.2
```

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: skaffold-operator.yaml [hunk 1]
```diff
@@ -1,253 +1,245 @@
...
-        - pkg/utils/retry
-        - pkg/utils/secrets
-        - pkg/utils/timewindow
-        - pkg/utils/validation/admissionplugins
-        - pkg/utils/validation/cidr
-        - pkg/utils/validation/features
-        - pkg/utils/version
-        - vendor
-        - VERSION
+    - image: eu.gcr.io/gardener-project/gardener/operator
```
Comment (rfranzke) [inline skaffold-operator.yaml]: Would it be possible to preserve the order of the keys and the indentation to shrink the diff?

File: skaffold-operator.yaml [hunk 2]
```diff
@@ -1,253 +1,245 @@
...
+          env[0].value: "true"
+          image.repository: '{{.IMAGE_REPO_eu_gcr_io_gardener_project_gardener_operator}}'
+          image.tag: '{{.IMAGE_TAG_eu_gcr_io_gardener_project_gardener_operator}}@{{.IMAGE_DIGEST_eu_gcr_io_gardener_project_gardener_operator}}'
+          imageVectorOverwrite: |
+            images:
+            - name: gardener-resource-manager
+              repository: localhost:5001/eu_gcr_io_gardener-project_gardener_resource-manager
+              tag: '{{.IMAGE_TAG_eu_gcr_io_gardener_project_gardener_resource_manager}}@{{.IMAGE_DIGEST_eu_gcr_io_gardener_project_gardener_resource_manager}}'
+          resource-manager.image.repository: '{{.IMAGE_REPO_eu_gcr_io_gardener_project_gardener_resource_manager}}'
+          resource-manager.image.tag: '{{.IMAGE_TAG_eu_gcr_io_gardener_project_gardener_resource_manager}}@{{.IMAGE_DIGEST_eu_gcr_io_gardener_project_gardener_resource_manager}}'
```
Comment (rfranzke) [inline skaffold-operator.yaml]: What is this needed for?

File: skaffold.yaml [hunk 3]
```diff
@@ -1,1145 +1,1111 @@
...
 ---
-apiVersion: skaffold/v2beta29
+apiVersion: skaffold/v4beta3
 kind: Config
 metadata:
   name: controlplane
 requires:
-- configs:
-  - etcd
+  - configs:
```
Comment (rfranzke) [inline skaffold.yaml]: same here

File: skaffold.yaml [hunk 4]
```diff
@@ -1,1145 +1,1111 @@
...
+          imageVectorOverwrite: |
+            images:
+            - name: gardenlet
+              repository: '{{.IMAGE_REPO_eu_gcr_io_gardener_project_gardener_gardenlet}}'
+              tag: '{{.IMAGE_TAG_eu_gcr_io_gardener_project_gardener_gardenlet}}@{{.IMAGE_DIGEST_eu_gcr_io_gardener_project_gardener_gardenlet}}'
+            - name: gardener-resource-manager
+              repository: '{{.IMAGE_REPO_eu_gcr_io_gardener_project_gardener_resource_manager}}'
+              tag: '{{.IMAGE_TAG_eu_gcr_io_gardener_project_gardener_resource_manager}}@{{.IMAGE_DIGEST_eu_gcr_io_gardener_project_gardener_resource_manager}}'
+          resource-manager.image.repository: '{{.IMAGE_REPO_eu_gcr_io_gardener_project_gardener_resource_manager}}'
+          resource-manager.image.tag: '{{.IMAGE_TAG_eu_gcr_io_gardener_project_gardener_resource_manager}}@{{.IMAGE_DIGEST_eu_gcr_io_gardener_project_gardener_resource_manager}}'
```
Comment (rfranzke) [inline skaffold.yaml]: same question here

File: skaffold-operator.yaml (hunk 2 above)
Comment (oliver-goetz) [reply skaffold-operator.yaml]: It was added by `skaffold fix` and I did not clean it up :sweat_smile:

File: skaffold.yaml (hunk 4 above)
Comment (oliver-goetz) [reply skaffold.yaml]: same as in `skaffold-operator.yaml`

**Review by rfranzke — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**